src/net_processing.cpp

```diff
@@ -3262,6 +3290,55 @@ bool PeerLogicValidation::SendMessages(CNode* pto, std::atomic<bool>& interruptM
             }
         }
 
+        // Check that outbound peers have reasonable chains
+        if (!(state.fProtectFromDisconnect || pto->fInbound || pto->m_manual_connection || pto->fFeeler || pto->fOneShot) && state.fSyncStarted) {
```

Comment:
Absorb `fInbound`, `m_manual_connection`, `fFeeler`, `fOneShot` into the method alluded to above.